types/tests/relationships-test.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  belongsTo,
+  BelongsTo,
+  hasMany,
+  HasMany,
+  Model,
+  Registry
+} from "miragejs";
+import Schema from "miragejs/orm/schema";
+
+const registryBelongsTo: BelongsTo<PersonRegistry> = belongsTo;
+const registryHasMany: HasMany<PersonRegistry> = hasMany;
+
+const PersonModel = Model.extend({
+  name: "hello",
+  parent: registryBelongsTo("person"),
+  pets: registryHasMany("pet"),
+  friends: registryHasMany<"pet" | "person">({ polymorphic: true })
+});
+
+const PetModel = Model.extend({
+  name: "fido",
+  owner: registryBelongsTo("person")
+});
+
+type PersonRegistry = Registry<
+  { person: typeof PersonModel; pet: typeof PetModel },
+  {}
+>;
+declare const schema: Schema<PersonRegistry>;
+
+const people = schema.all("person");
+
+people.length; // $ExpectType number
+people.modelName; // $ExpectType string
+people.models.map(model => {
+  model.parent!.name; // $ExpectType string
+  model.parent!.parent!.name; // $ExpectType string
```

Comment:
You might want to invert this and make it `model.parent?.name` expecting type `string | null` (or `string | undefined`, whichever is correct) and the same for the two-parent lookup. We strongly prefer to *handle* optionals rather than to *assert over them*. Here it doesn't matter so much, as this still correctly tests the types, so you don't have to change it, but it'd be my recommendation. Code like this tends to get read by the curious and used as an example.

Reply:
TypeScript updated to 3.8, minimum required TS is also updated to 3.7 in dts header. Now we can use `?` syntax, so the test is updated. Thanks for the suggestion.